home_2/hometask/src/Weapon.ts

```diff
@@ -0,0 +1,48 @@
+import { Item } from "./Item";
+
+export abstract class Weapon extends Item {
+  static MODIFIER_CHANGE_RATE: number = 0.05;
+  private baseDurability: number;
+  protected baseDamage: number;
+  protected damageModifier: number = 0;
+  protected durabilityModifier: number = 0;
+  private isBroken: boolean = false;
+
+  constructor(name: string, baseDamage: number, baseDurabitity: number, value: number, weight: number) {
+    super(name, value, weight);
+    this.baseDamage = baseDamage;
+    this.baseDurability = baseDurabitity;
+  }
+
+  use(): string {
+    if (this.isBroken) {
+      return `You can't use the ${this.name}, it is broken.`;
+    }
+    this.baseDurability = this.getEffectiveDurability(-Weapon.MODIFIER_CHANGE_RATE);
+    const additionalText = this.baseDurability === 0 ? `\nThe ${this.name} breaks.` : "";
+    this.isBroken = true;
+    return `You use the ${this.name}, dealing ${Weapon.MODIFIER_CHANGE_RATE} points of damage.${additionalText}`;
+  }
+
+  polish(): void {}
```

Comment:
if the method is expected to be overridden in the abstract class - it's better to mark it abstract as well: 
`abstract polish(): void;`
This will require any descendent to implement - so it won't be left without attention when ingerited